backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/AddVmCommand.java

```diff
@@ -1118,6 +1126,13 @@ private void addGraphicsDevice() {
         }
     }
 
+    private void addMdevDevices() {
+        if (mdevType != null) {
+            var convertedMdevDevices = MDevTypesUtils.convertDeprecatedCustomPropertyToVmDevices(mdevType, getVmId());
+            vmDeviceDao.saveAll(convertedMdevDevices);
+        }
+    }
```

Comment:
looking on all other devices, many are having its own command with validation part. don't you need to validate anything for the mdev?
same comment for update vm.

Reply:
right, there are such comment (for example, AddMdevCommand) but this code is for backward compatibility with clients that keep using the custom property mdev_type and we assume that the property is properly set (new clients should rather use the new method for adding/updating the mediated devices)

Reply:
After offline discussion, we can use the commands but the validation, or better say the most of the code is redundant. Here we create the device, so we just need to save it into the DB.